File: 2021/Day1/Day1.scala
import scala.io.Source
@main def day1 =
    val input = Source
        .fromFile("input.txt")
        .getLines
        .map(_.toInt)
        .toList

    val part1 = input
        .sliding(2)
        .map(pair => (pair(0), pair(1)))
        .count(_ < _)

    val part2 = input
        .sliding(3)
        .map(_.sum)
        .sliding(2)
        .map(pair => (pair(0), pair(1)))
        .count(_ < _)

    println(s"Part 1: ${part1}")
    println(s"Part 2: ${part2}")

File: 2021/Day1/Day1.fs
module Day1

open System.IO

let input = File.ReadAllLines "input.txt" |> Array.map int

let deeper count (prev, curr) =
    if curr > prev then count + 1
    else count

let fisrtPart input =
    input
        |> Array.pairwise
        |> Array.fold deeper 0

let secondPart input =
    input
        |> Array.windowed 3
        |> Array.map Array.sum
        |> Array.pairwise
        |> Array.fold deeper 0

fisrtPart input |> printfn "First %d"
secondPart input |> printfn "Second %d"
